### PR TITLE
Umip-86 - Add NGNUSD, PHPUSD, CADUSD, ZARUSD, KRWUSD and JPYUSD as price identifiers

### PR DESCRIPTION
**Pull request by SayWhatDev**

### CODE REVIEW

**Review by therealhexi — COMMENTED** [review]
Initial comments

File: UMIPs/umip-86.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+The DVM should support price requests for the NGNUSD, PHPUSD, CADUSD, ZARUSD, KRWUSD and JPYUSD price index.
+
+## Motivation
+The DVM currently does not support these Forex price index.
+
+### Cost: 
+While Forex data are free through open centralized exchange, brokers, and other sources APIs, the most reliable are paid. We propose to use TraderMade's APIs whose pricing is accessible on their website. TraderMade has also a Free Tier which can be used by the voters and would be sufficient to provide the price of a certain asset.
+
+### Opportunity: 
+Synthetic tokens that track Forex pairs such as CADUSD and JPYUSD could be used both for speculations and hedging, but we see a bigger opportunity for using them as building blocks for helping other DeFi protocols and dApp addressing the global market.
```
[the remark this answers is not in the document]
Comment (Pascaltallarida) [reply UMIPs/umip-86.md:19]: > Interesting. Could you elaborate on how DeFi protocols could leverage this to address the global market?

Hello,

These price identifiers will be used in Jarvis Network, an approved forked of UMA which allows to the minting of perpetual synthetic asset, without any interests/stability fee etc.

This allows to use them as a building block and to create fiat on/off-ramp for them.

In this particular case, these synths can be used in Aave for example, Uniswap, or within other protocols to give them more utility/yield/liquidity. Also, our protocol is a mixed between the UMA's self minting approach (we call it the bank contract) and Synthetix's (we call it the broker contract) which uses Chainlink to make our synth redeemable for a USD-stablecoin); the latter feature allows to burn any Synths for USDC, at the oracle price, and then to exchange these USDC for wBTC for ex on Uniswap, in the same transaction. So we can exchange for ex our synthetic NGN for wBTC at the BTCUSD Uniswap's rate.

Our goal is exactly this. Providing any user with the synthetic equivalence of their local fiat currency.

You can read more about it here: [https://medium.com/jarvis-network/a-multi-layer-ecosystem-of-synthereum-and-our-go-to-market-strategy-56d5daa9b709](https://medium.com/jarvis-network/a-multi-layer-ecosystem-of-synthereum-and-our-go-to-market-strategy-56d5daa9b709)

File: UMIPs/umip-86.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+- Rounding: Closest, 0.5 up (> .000005 rounds up, < .000005 rounds down)
+- Scaling Decimals: 18 (1e18)
+- Pricing Interval: 60 seconds
+- Dispute timestamp rounding: down
+
+## Rationale
+Apart from the weekend, there is little to no difference in prices on liquid major Forex pairs like JPYUSD, so any price feed could be used; however, for convenience, we recommend using the one of TraderMade.
+
+## Implementation
+Historical PHPUSD, CADUSD, ZARUSD, KRWUSD and JPYUSD prices from TraderMade are available on minute increments. Historical NGNUSD price from TraderMade are not accessible before April, 30th 2021 (not issue). Price requests should use the minute price that is nearest and earlier than the price request timestamp. To do this, voters should use the open price of the OHLC period that the price request timestamp falls in. TraderMade endpoints are queried based on the OHLC period's close time.
```
[the remark this answers is not in the document]
Comment (Pascaltallarida) [reply UMIPs/umip-86.md:78]: Thx, we will fix this

File: UMIPs/umip-86.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+- Base Currency: JPY
+- Quote Currency: USD
+
+-----------------------------------------
+
+- Source: https://marketdata.tradermade.com
+- Result Processing: None
+- Input Processing: None. Human intervention in extreme circumstances where the result differs from broad market consensus.
+- Price Steps: 0.00001 (5 decimals in more general trading format)
+- Rounding: Closest, 0.5 up (> .000005 rounds up, < .000005 rounds down)
```
[the remark this answers is not in the document]
Comment (Pascaltallarida) [reply UMIPs/umip-86.md:69]: Could you develop more?

[the remark this answers is not in the document]
Comment (SayWhatDev) [reply UMIPs/umip-86.md]: This issue has been address in the following [commit](7a3f77d4e2cad89fa6cb8f0c7bb9f36b8dd2f15d)

File: UMIPs/umip-86.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (SayWhatDev) [reply UMIPs/umip-86.md]: This issue has been addressed in the following [commit](https://github.com/UMAprotocol/UMIPs/pull/273/commits/7a3f77d4e2cad89fa6cb8f0c7bb9f36b8dd2f15d)

**Pull request closed without merging**